platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/MerkleDbCompactionCoordinator.java

```diff
@@ -133,26 +124,24 @@ public void resumeCompaction() throws IOException {
      * Stops all compactions in progress and disables background compaction. All subsequent calls to
      * compacting methods will be ignored until {@link #enableBackgroundCompaction()} is called.
      */
-    void stopAndDisableBackgroundCompaction() {
-        compactionEnabled.set(false);
+    synchronized void stopAndDisableBackgroundCompaction() {
```

Comment:
I don't see why `stopAndDisableBackgroundCompaction` should not be `public` like `pauseCompaction`/`resumeCompaction`.

Reply:
OK, I changed pauseCompaction() and resumeCompaction() to package private, too